src/containers/CreateCustomRun/CreateCustomRun.jsx

```diff
@@ -0,0 +1,484 @@
+      const newInvalidLabels = { ...prevState[invalidProp] };
+      const removedLabel = newLabels[index];
+      newLabels.splice(index, 1);
+      if (removedLabel.id in newInvalidLabels) {
+        delete newInvalidLabels[`${removedLabel.id}-key`];
+        delete newInvalidLabels[`${removedLabel.id}-value`];
+      }
+      return {
+        ...prevState,
+        [prop]: newLabels,
+        [invalidProp]: newInvalidLabels
+      };
+    });
+  }
+
+  function handleChangeLabel(prop, invalidProp, { type, index, value }) {
+    setState(prevState => {
+      const newLabels = [...prevState[prop]];
+      newLabels[index][type] = value;
+      const newInvalidLabels = { ...prevState[invalidProp] };
+      if (!isValidLabel(type, value)) {
+        newInvalidLabels[`${newLabels[index].id}-${type}`] = true;
+      } else {
+        delete newInvalidLabels[`${newLabels[index].id}-${type}`];
+      }
+      return {
+        ...prevState,
+        [prop]: newLabels,
+        [invalidProp]: newInvalidLabels
+      };
+    });
+  }
+
+  function handleCloseYAMLEditor() {
+    let url = urls.customRuns.all();
+    if (defaultNamespace && defaultNamespace !== ALL_NAMESPACES) {
+      url = urls.customRuns.byNamespace({ namespace: defaultNamespace });
+    }
+    navigate(url);
+  }
+
+  function handleCreate({ resource }) {
+    const resourceNamespace = resource?.metadata?.namespace;
+    return createCustomRunRaw({
+      namespace: resourceNamespace,
+      payload: resource
+    }).then(() => {
+      navigate(urls.customRuns.byNamespace({ namespace: resourceNamespace }));
+    });
+  }
+
+  function handleNamespaceChange({ selectedItem }) {
+    const { text = '' } = selectedItem || {};
+    if (text !== namespace) {
+      setState(state => ({
+        ...state,
+        ...initialState,
+        kind: state.kind,
+        namespace: text
+      }));
+
+      const queryParams = new URLSearchParams(location.search);
+      if (text) {
+        queryParams.set('namespace', text);
+      } else {
+        queryParams.delete('namespace');
+      }
+      queryParams.delete('customName');
+      const browserURL = location.pathname.concat(`?${queryParams.toString()}`);
+      navigate(browserURL);
+    }
+  }
+
+  function handleKindChange({ selectedItem }) {
+    const { text = '' } = selectedItem || {};
+    if (text !== kind) {
+      setState(state => ({
+        ...state,
+        ...initialState,
+        kind: text
+      }));
+
+      const queryParams = new URLSearchParams(location.search);
+      queryParams.set('kind', text);
+      queryParams.delete('namespace');
+      queryParams.delete('customName');
+      const browserURL = location.pathname.concat(`?${queryParams.toString()}`);
+      navigate(browserURL);
+    }
+  }
+
+  function handleParamChange(key, value) {
+    setState(state => ({
+      ...state,
+      params: {
+        ...state.params,
+        [key]: value
+      }
+    }));
+  }
```

Comment:
Not used.

There may be more. Please review any remaining functions to ensure they're actually called, or remove them if they're no longer needed.